cartography/intel/aws/secretsmanager.py

```diff
@@ -76,6 +81,115 @@ def cleanup_secrets(neo4j_session: neo4j.Session, common_job_parameters: Dict) -
     )
 
 
+@timeit
+@aws_handle_regions
+def get_secret_versions(
+    boto3_session: boto3.session.Session, region: str, secret_arn: str
+) -> List[Dict]:
+    """
+    Get all versions of a secret from AWS Secrets Manager.
+    """
+    client = boto3_session.client("secretsmanager", region_name=region)
+    versions: List[Dict] = []
+    try:
+        paginator = client.get_paginator("list_secret_version_ids")
+        for page in paginator.paginate(SecretId=secret_arn, IncludeDeprecated=True):
+            for version in page["Versions"]:
+                # Add SecretId to establish the parent-child relationship
+                version["SecretId"] = secret_arn
+                # Format ARN to match expected format: {secret_arn}:version:{version_id}
+                version["ARN"] = f"{secret_arn}:version:{version['VersionId']}"
+
+                # Convert createdDate to a datetime object if it's not already
+                if "CreatedDate" in version and not isinstance(
+                    version["CreatedDate"], datetime.datetime
+                ):
+                    version["CreatedDate"] = datetime.datetime.fromtimestamp(
+                        version["CreatedDate"], tz=datetime.timezone.utc
+                    )
+
+            versions.extend(page["Versions"])
+
+    except client.exceptions.ResourceNotFoundException:
```

Comment:
No need to handle these exceptions here because you're using the decorator -- no need to use any `try`+`except`.